### PR TITLE
[IMP] *: remove guarded catch

### PR DESCRIPTION
**Pull request by mcm-odoo**
This commit removes the promise extension which added the function `guardedCatch`. It was used to filter server and connection errors from javascript errors. Instead of using guardedCatch, we should use catch and check if the reason is a server or connection error if needed and re-throw the error if it's not handled.

task: 3439226
Enterprise: https://github.com/odoo/enterprise/pull/48451

### CODE REVIEW

**Review by AntoineVDV — APPROVED** [review]
Good stuff! I assume this will allow using try/catch clauses instead of chaining?

Also, your ENT PR is not mentioned in your COM PR (and vice-versa), which makes it harder to discover.

**Comment by mcm-odoo** [comment]
> I assume this will allow using try/catch clauses instead of chaining?

try/catch is already allowed but yes I think it would be easier to use it when `guardedCatch` is removed.

> Also, your ENT PR is not mentioned in your COM PR (and vice-versa), which makes it harder to discover.

done.

File: addons/web/static/src/core/network/rpc_service.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ export class ConnectionLostError extends Error {
 
 export class ConnectionAbortedError extends Error {}
 
+export function isRequestError(e) {
+    return e instanceof RPCError || e instanceof ConnectionLostError;
```
Comment (aab-odoo) [inline addons/web/static/src/core/network/rpc_service.js:31]: Can't we let the ConnectionLost handler deal with those errors? I'm wondering whether we couldn't only check for RPCErrors everywhere you use this function.

Also, currently, I think the name of the function is a bit vague.

File: addons/web/static/src/core/errors/error_service.js [hunk 2]
```diff
@@ -110,6 +110,10 @@ export const errorService = {
 
         browser.addEventListener("unhandledrejection", async (ev) => {
             const error = ev.reason;
+            if (typeof error !== "string" && !(error instanceof Error)) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/errors/error_service.js:113]: why not string?

**Review by aab-odoo — COMMENTED** [review]
wow that's a big milestone :+1:

File: addons/web/static/src/core/errors/error_service.js (hunk 2 above)
Comment (mcm-odoo) [reply addons/web/static/src/core/errors/error_service.js:113]: We have a test that checks it handles strings. IMO this is not correct, only `Error` objects should be handled here.

Comment (aab-odoo) [reply addons/web/static/src/core/errors/error_service.js:113]: I agree, which test?

File: addons/web/static/src/core/network/rpc_service.js (hunk 1 above)
Comment (mcm-odoo) [reply addons/web/static/src/core/network/rpc_service.js:31]: I tried to reproduce the `guardedCatch` check this function to have the minimal difference but yes I think we can only check for `RPCError` and let error service handle the `ConnectionLostError`

File: addons/web/static/src/core/errors/error_service.js (hunk 2 above)
Comment (mcm-odoo) [reply addons/web/static/src/core/errors/error_service.js:113]: "can handle rejected promise errors with a string as reason" in `error_service_tests.js:66`

Comment (aab-odoo) [reply addons/web/static/src/core/errors/error_service.js:113]: Filtering out the string case here is only relevant to make the test pass, but according to the [description of the commit adding the test](https://github.com/odoo/odoo/commit/5d45465758ae46e1838a97ca5c7aeee47d9256c0), simply ignoring rejection of strings would do the trick as well to fix the issue, so I think I would remove it.

Would be nice to have @sdegueldre input on this PR anyway.

Comment (sdegueldre) [reply addons/web/static/src/core/errors/error_service.js:113]: I don't think we should be ignoring things that are thrown even if they're not errors. This just silently eats real crashes.

**Comment by ged-odoo** [comment]
robodoo  override=ci/security
It is just code reindented. Also, it's a markup of something that has been escaped already

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**